main.py

```diff
@@ -109,23 +115,21 @@ def load_chunks(inventory, lg, ck):
   return chks
 
 
-def load_vectors(docs):
+def load_vector(d):
   embed = OpenAIEmbeddings()
-  msg = f"Loading vectors..."
-  sz = len(docs)
-  prog = st.progress(0.0, text=msg)
-  vectors = []
-  for i, d in enumerate(docs, start=1):
-    vectors.append(embed.embed_query(d.page_content))
-    prog.progress(i/sz, text=msg)
-  prog.empty()
-  return vectors
+  result = embed.embed_query(d.page_content)
+  return result
```

Comment:
I think we can we can return immediately without unnecessarily assigning it to yet another variable.

```suggestion
  return embed.embed_query(d.page_content)
```